storage/cache/subtree_cache.go

```diff
@@ -174,8 +172,8 @@ func (s *SubtreeCache) preload(ids []tree.NodeID, getSubtrees GetSubtreesFunc) e
 	// Create empty subtrees for anything left over. Note that multiple parallel readers
 	// may be be running this code and touch the same keys, although this doesn't happen
 	// normally.
-	for _, id := range want {
-		if err := s.cacheSubtree(s.newEmptySubtree(id)); err != nil {
+	for id := range want {
+		if err := s.cacheSubtree(s.newEmptySubtree([]byte(id))); err != nil {
```

Comment:
May have been true before but there's no test covering this line.

Reply:
Yes, it [wasn't](https://codecov.io/gh/google/trillian/src/master/storage/cache/subtree_cache.go) covered previously (same for other lines, this PR seemingly shouldn't change coverage of `subtree_cache.go`).

Unlike for maps, this line won't be executed for logs, because in this spot logs read only existing tiles on the right border, so `len(want)` will end up zero (cleared by `delete` statement a few lines above). In fact, it would probably be safer to return an error here instead of making an empty tile (which in fact might be not empty but lost due to a bug).

#2406